api4/cloud.go

```diff
@@ -116,6 +116,10 @@ func changeSubscription(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getSubscriptionStats(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleReadBilling) {
+		c.SetPermissionError(model.PermissionSysconsoleReadBilling)
+		return
+	}
```

Comment:
We want all authenticated users to be able to get these stats not just those who can see the system console. To fix the bug you probably want to change: 
https://github.com/mattermost/mattermost-server/pull/19811/files#diff-5b4abeaf1bd0792473cef52eb3e3022b373d62124be75c83676027f10d381b0aR41

To have a ` api.APISessionRequired` instead of the generic, unauthenticated `api.APIHandler`